### PR TITLE
Surface game metadata and player nicknames in client / react props

### LINKED ISSUE #413: surface player nicknames in client
This is available in the player metadata section [here](https://github.com/nicolodavis/boardgame.io/blob/master/src/server/api.ts#L172) and should be surfaced to the game client.

### PR DESCRIPTION
**Pull request by jasonharrison**
Will close #413 
#### Checklist

* [X] Use a separate branch in your local repo (not `master`).
* [X] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

File: src/master/master.js [hunk 1]
```diff
@@ -301,7 +302,7 @@ export class Master {
     this.transportAPI.send({
       playerID,
       type: 'sync',
-      args: [gameID, filteredState, log],
+      args: [gameID, filteredState, log, gameMetadata],
```
Comment (nicolodavis) [inline src/master/master.js]: We should scrub the credentials from `gameMetadata` before it is sent over.

Comment (jasonharrison) [reply src/master/master.js]: I overlooked this, you are right.

File: src/master/master.js [hunk 2]
```diff
@@ -259,14 +259,20 @@ export class Master {
   async onSync(gameID, playerID, numPlayers) {
     const key = gameID;
 
-    let state;
+    let state, gameMetadata, filteredGameMetadata;
 
     if (this.executeSynchronously) {
       state = this.storageAPI.get(key);
+      gameMetadata = await this.storageAPI.get(GameMetadataKey(gameID));
```
Comment (nicolodavis) [inline src/master/master.js]: I think you want to `await` in the `else` block, not here.

File: src/client/transport/local.js [hunk 3]
```diff
@@ -121,6 +121,8 @@ export class Local {
    */
   subscribe() {}
 
+  subscribeGameMetadata() {}
```
Comment (nicolodavis) [inline src/client/transport/local.js]: Change line 89 in this file to also reflect the new `gameMetadata` argument that's sent back (even if it doesn't really do anything in the local mode).

**Comment by nicolodavis** [comment]
Thanks @jasonharrison.  Can you also double-check that this doesn't break anything when the client is used in a singleplayer mode (i.e. no lobby or multiplayer)?

**Comment by haoyangnz** [comment]
@nicolodavis Are we able to release a patch that includes this? Or do we have some fixed release cycle or testing to do?
Sorry for all the questions 🙏

**Comment by nicolodavis** [comment]
The release will include everything until the tip in `master`.  I want to hold off on that until I get the docs ready for all the new features that have changed the API regarding phases.

One option is you can use the library from the repo directly by using `npm link`.  That way you don't have to wait for a release.

**Comment by haoyangnz** [comment]
Alright 👍

**Pull request merged**